enterprise/cluster/src/main/java/org/neo4j/cluster/protocol/heartbeat/HeartbeatState.java

```diff
@@ -229,8 +229,8 @@ public HeartbeatState handle( HeartbeatContext context,
                             context.getInternalLog( HeartbeatState.class )
                                     .debug( "Received suspicions as " + suspicions );
 
-                            URI from = new URI( message.getHeader( Message.FROM ) );
-                            InstanceId fromId = context.getIdForUri( from );
+                            InstanceId fromId = new InstanceId(Integer.parseInt(message.getHeader( Message.INSTANCE_ID )));
```

Comment:
Same comment here as before about backwards compatibility. This should also check against URIs